otx/core/data/adapter/segmentation_dataset_adapter.py

```diff
@@ -52,3 +66,104 @@ def get_otx_dataset(self) -> DatasetEntity:
 
         self.remove_unused_label_entities(used_labels)
         return DatasetEntity(items=dataset_items)
+
+
+class SelfSLSegmentationDatasetAdapter(SegmentationDatasetAdapter):
+    """Self-SL for segmentation adapter inherited from SegmentationDatasetAdapter."""
+
+    # pylint: disable=protected-access
+    def _import_dataset(
+        self,
+        train_data_roots: Optional[str] = None,
+        val_data_roots: Optional[str] = None,
+        test_data_roots: Optional[str] = None,
+        unlabeled_data_roots: Optional[str] = None,
+        pseudo_mask_dir: str = "detcon_mask",
+    ) -> Dict[Subset, DatumaroDataset]:
+        """Import custom Self-SL dataset for using DetCon.
+
+        Self-SL for semantic segmentation using DetCon uses pseudo masks as labels,
+        but Datumaro cannot load this custom data structure because it is not in Datumaro format.
+        So, it is required to manually load and set annotations.
+
+        Args:
+            train_data_roots (Optional[str]): Path for training data.
+            pseudo_mask_dir (str): Directory to save pseudo masks. Defaults to "detcon_mask".
+
+        Returns:
+            DatumaroDataset: Datumaro Dataset
+        """
+        assert train_data_roots
+        logger = get_logger()
+        logger.warning(f"Please check if {train_data_roots} is data roots only for images, not annotations.")
+
+        dataset = {}
+        dataset[Subset.TRAINING] = DatumaroDataset.import_from(train_data_roots, format="image_dir")
+        self.is_train_phase = True
+
+        # Load pseudo masks
+        img_dir = None
+        total_labels = []
+        flag_create_mask = False
+        for item in dataset[Subset.TRAINING]:
+            img_path = item.media.path
+            if img_dir is None:
```

Comment:
As far as I understand, `img_dir` is always None. Is this `if`` statement necessary?

Reply:
I inserted if statement to update `img_dir` in the first loop.
`img_dir` is updated only in the first loop, and it is not updated afterwards and is used in the function.